cmake/version.cmake

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+# If possible, deduce project version from git environment.
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
+  find_package(Git)
+
+  execute_process(
+    COMMAND git describe --tags
```

Comment:
After the `find_package`, can we wrap all of this in `if (Git_FOUND)` and use `Git::Git describe ...` rather than `git describe ...`? Again, a real minor case (you're in a `.git` directory but don't have git installed?), but it seems better etiquette to delegate to CMake (maybe git is installed, but deliberately unavailable to CMake?).